### PR TITLE
feat(LERN-49): Implemented getAllSchoolsForAdmin endpoint.

### PR DESCRIPTION
**Pull request by SleepyPeel**

### CODE REVIEW

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt [hunk 1]
```diff
@@ -16,4 +18,6 @@ public class SchoolResource(private val schoolService: SchoolService) {
     @PostMapping("/create")
     public fun createSchool(@RequestBody createSchoolDto: CreateSchoolDto): Unit =
         schoolService.createSchool(createSchoolDto)
+    @GetMapping("/getAllSchoolsForAdmin")
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:21]:
```suggestion
    @GetMapping(params = ["assignedUserId"])
```

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt [hunk 2]
```diff
@@ -16,4 +18,6 @@ public class SchoolResource(private val schoolService: SchoolService) {
...
+    public fun getAllSchoolsForAdmin(@RequestParam userId: UUID): List<SchoolDto> = schoolService.getAllSchools()
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:22]:
```suggestion
    public fun getSchoolsByUser(@RequestParam userId: UUID): List<SchoolDto> = schoolService.getAllSchools()
```

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolService.kt [hunk 3]
```diff
@@ -24,6 +24,10 @@ public class SchoolService(
 
     public fun getAllSchools(): List<SchoolDto> = schoolDao.fetchAllSchools().map(::mapToDto)
 
+    public fun getAllSchoolsForAdmin(userId: UUID): List<SchoolDto> = schoolDao.fetchAllSchoolsForAdmin(userId).map(
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolService.kt:27]:
```suggestion
    public fun getSchoolsByUserId(userId: UUID): List<SchoolDto> = schoolDao.fetchAllSchoolsForAdmin(userId).map(
```

**Review by deepspace1000 — CHANGES_REQUESTED** [review]
Nice job. mostly some renaming and change of the @GetMapping endpoint.

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt (hunk 1 above)
Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:21]: @A1eeee Have we used the params like this before here?

Comment (A1eeee) [reply backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:21]: No, we never used it the query params like that before.

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt (hunk 2 above)
Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:22]: FYI you called the wrong schoolService function it's not getAllSchools but getAllSchoolsForAdmin (which should also be renamed.

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt (hunk 1 above)
Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/backoffice/school/SchoolResource.kt:21]: Okay, there isn't a real difference in doing it like this but documenting it in the @GetMapping annotation. so i would go with my suggestion.

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Pull request merged**